clang/tools/driver/CacheLauncherMode.cpp

```diff
@@ -105,6 +105,15 @@ static void addCommonArgs(bool ForDriver, SmallVectorImpl<const char *> &Args,
       }
     }
   };
+  auto RemotePath =
+      llvm::sys::Process::GetEnv("LLVM_CACHE_REMOTE_SERVICE_SOCKET_PATH");
+
+  if (!llvm::sys::Process::GetEnv("DISABLE_MCCAS") && !RemotePath) {
+    addCC1Args({"-fcas-backend"});
```

Comment:
These should go to `addLauncherArgs`, `addCommonArgs` is for flags common to launcher and dep-scan server.